src/PRAS2PowerSystems.jl

```diff
@@ -1,240 +1,265 @@
-    asset_dict = Dict([
-        (:generators, (gens_avail, PSY.Generator)),
-        (:storages, (stors_avail, PSY.Storage)),
-        (:generatorstorages, (gen_stors_avail, PSY.StaticInjection)),
-        (:lines, (lines_avail, PSY.Branch)),
-    ])
-    asset_keys = []
-
-    for k in keys(asset_dict)
-        if (length(getfield(pras_system, k)) != 0)
-            push!(asset_keys, k)
-        end
-    end
-    @info "Scenarios of interest will be sorted according to sample unserved energy. The PSY systems for all the scenarios will be exported using to_json() method in InfrastructureSystems."
-
-    sample_idx = sortperm(shortfall_samples[], rev=true)
-    resolution = Dates.Hour(1)
-    for i in 1:num_scenarios
-        for k in asset_keys
-            asset_status_all = asset_dict[k][1].available[:, :, sample_idx[i]]
-            asset_names = getfield(asset_dict[k][1], k)
-            for (j, asset_name) in enumerate(asset_names)
-                # Creating TimeSeries Data
-                timestamps = range(
-                    Dates.DateTime(pras_system.timestamps.start),
-                    step=resolution,
-                    length=length(pras_system.timestamps),
-                )
-                availability_data = TimeSeries.TimeArray(timestamps, asset_status_all[j, :])
-                availability_timeseries =
-                    PSY.SingleTimeSeries("availability", availability_data)
-                #Removing TimeSeries Data associated with the component if it already exists
-                if (i != 1)
-                    PSY.remove_time_series!(
-                        psy_sys,
-                        PSY.SingleTimeSeries,
-                        PSY.get_component(asset_dict[k][2], psy_sys, asset_name),
-                        "availability",
-                    )
-                end
-                # Adding TimeSeries Data to PSY System
-                PSY.add_time_series!(
-                    psy_sys,
-                    PSY.get_component(asset_dict[k][2], psy_sys, asset_name),
-                    availability_timeseries,
-                )
-            end
-        end
+Analyze resource adequacy using Monte Carlo simulation and add the asset status from the worst sample
+to PSY.TimeSeriesForcedOutage of the component.
 
-        file_name = dir_name * "/$i.json"
-        PSY.IS.to_json(psy_sys, file_name, force=false)
-        @info "Succesfully exported PSY System with outage profile for scenario $(i)."
-    end
-    @info "Succesfully exported PSY System(s). The .json files and corresponding data .h5 files for all scenarios are available here : $(dir_name)."
+# Arguments
+
+  - `sys::PSY.System`: PowerSystems.jl system model
+  - `template::RATemplate`: PRAS problem template
+  - `method::PRASCore.SequentialMonteCarlo`: Simulation method to use
+
+# Returns
+
+ - PSY System with PSY.TimeSeriesForcedOutage for all components for which asset status is available
+"""
+function generate_outage_profile!(
```

Comment:
Should we have a function that takes a PRAS system still?

Reply:
Probably not. The PRAS system should be hidden from the user.

Reply:
Also, you finally need a PSY System to add this time series to. This will make things messy and we have no control of mapping from the PRAS System and PSY System the user is passing.